[Feature] Add the unit test to test the `scale_factor` in LetterResize

tang576225574 opened the pull request:
## Motivation
之前在 #387 的 PR 中提到需要添加 unit test，但没来得及提交和回复，现在补上（基于新增的scale_factor的unit test进行了修改）


## Modification

只改动了 `tests/test_datasets/test_transforms/test_transforms.py`，修改了scale_factor的unit test

hhaAndroid commented on tests/test_datasets/test_transforms/test_transforms.py:
```diff
@@ -125,23 +125,35 @@ def test_letter_resize(self):
         transform = [
-            YOLOv5KeepRatioResize(scale=(32, 32)),
-            LetterResize(scale=(64, 68))
+            YOLOv5KeepRatioResize(scale=(1280, 1280)),
```
1280太大了，单元测试的核心在于测试功能，如果非必要，不建议弄这么大的图片

tang576225574 replied:
好的

tang576225574 commented:
已更新；同时换了一种验证方式，之前写的验证方式存在问题，现在改成验证scale_facter和pad_param的值的正确性；另外发现transforms.py文件里letterresize存在问题：有两处地方height和width的位置调反了，虽然不影响结果（因为scale_facter的height和width值相同），但是含义不对会引起confuse，已修改过来。

I have updated the branch. note: I changed another way to validate the unittest because something was wrong in the original validation method. Another change is that I corrected two place in letterresize which was confuse by the definition of  image height and width.

hhaAndroid commented on mmyolo/datasets/transforms/transforms.py:
```diff
@@ -240,7 +240,7 @@ def _resize_img(self, results: dict):
         if 'pad_param' in results:
             results['pad_param_origin'] = results['pad_param'] * \
-                np.repeat(scale_factor, 2)
+                np.repeat(ratio, 2)
```
Is there a difference between scale_factor and ratio?

tang576225574 commented:
能合并吗？还是说需要修改？

hhaAndroid commented:
> 

我留了一个评论，你回复下？
